File: public/inject_search.py
import os
import glob

directory = "."
script_tag = '\n<script src="/search_redirect.js"></script>\n'
count = 0

# Find all HTML files recursively
files = glob.glob(os.path.join(directory, '**', '*.html'), recursive=True)
print(f"Found {len(files)} HTML files.")

for filepath in files:
    try:
        # Skip search_redirect.js itself if it was somehow found (though glob is for .html)
        if not filepath.endswith('.html'):
            continue

        with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
            content = f.read()
        
        if 'search_redirect.js' in content:
            continue
        
        if '</body>' in content:
            new_content = content.replace('</body>', f'{script_tag}</body>')
        else:
            new_content = content + script_tag
            
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        count += 1
        
        if count % 100 == 0:
            print(f"Processed {count} files...")
            
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

print(f"Successfully injected search script into {count} files.")

File: public/inject_search.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(fullPath));
        } else { 
            if (file.endsWith('.html')) {
                results.push(fullPath);
            }
        }
    });
    return results;
}

const directory = ".";
console.log(`Scanning directory: ${path.resolve(directory)}`);
const files = walk(directory);
console.log(`Found ${files.length} HTML files.`);

let count = 0;
const scriptTag = '\n<script src="/search_redirect.js"></script>\n';

files.forEach((file, index) => {
    try {
        const content = fs.readFileSync(file, 'utf8');
        
        if (content.includes('search_redirect.js')) {
            return;
        }

        if (content.includes('</body>')) {
            const newContent = content.replace('</body>', `${scriptTag}</body>`);
            fs.writeFileSync(file, newContent, 'utf8');
            count++;
        } else {
            fs.appendFileSync(file, scriptTag, 'utf8');
            count++;
        }

        if ((index + 1) % 50 === 0 || index === files.length - 1) {
            console.log(`Processed ${index + 1} / ${files.length} files...`);
        }
    } catch (e) {
        console.error(`Error processing ${file}: ${e}`);
    }
});

console.log(`Successfully injected search script into ${count} files.`);
